dev-requirements.txt

```
@@ -21,7 +21,8 @@ sphinx-rfcsection~=0.1.1
 # (or until py3.9 EOL... in 10/2025, I HOPE NOT)
 vcrpy @ git+https://github.com/sopel-irc/vcrpy@uncap-urllib3
 # type check
-mypy>=1.3,<2
+# often breaks CI on master, so pin and update deliberately, on our own terms
+mypy==1.11.2
```

Nitpick: IMO it would be okay to allow patch versions in, but I don't feel super strongly about it.

```suggestion
mypy>=1.11, <1.12
```